### PR TITLE
Fix `mortonEncode` `use-of-uninitialized-value`

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

https://s3.amazonaws.com/clickhouse-test-reports/0/d86eae3dd9f6a85c4db1895390bb4c46396ed431/ast_fuzzer__msan_.html

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 9e1d8b8aaa438d7280c10f966102fab2aa1ae46b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62288/9e1d8b8aaa438d7280c10f966102fab2aa1ae46b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Functions/mortonEncode.cpp [hunk 1]
```diff
@@ -321,6 +321,9 @@ class FunctionMortonEncode: public TargetSpecific::Default::FunctionMortonEncode
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
     {
+        if (input_rows_count == 0)
```
Comment (alexey-milovidov) [inline src/Functions/mortonEncode.cpp:324]: Ok.

I see a way to automate it.

Shouldn't all functions return something like `return_type->createColumn()` for empty blocks?

If there are special cases, we can make `useDefaultImplementationForEmptyBlocks` :)

**Review by alexey-milovidov — APPROVED** [review]

File: src/Functions/mortonEncode.cpp (hunk 1 above)
Comment (antonio2368) [reply src/Functions/mortonEncode.cpp:324]: > Shouldn't all functions return something like return_type->createColumn() for empty blocks?

Most likely.
I'll add it to `IExecutableFunction::execute` and see what happens.

**Comment by clickgapai** [comment]
Hi @antonio2368 @alexey-milovidov — while reviewing this PR I found the following:

- **Coverage gap:** PR's `input_rows_count == 0` early-return branch in `mortonEncode` dispatcher is untested — https://github.com/ClickHouse/ClickHouse/pull/104010

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**